data_catalog_backend/routes/provider_routes.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from typing import List
+import uuid
+from fastapi import APIRouter, Depends, HTTPException
+
+from data_catalog_backend.dependencies import get_provider_service
+from data_catalog_backend.models import Provider
+from data_catalog_backend.schemas.provider import ProviderRequest, ProviderResponse
+from data_catalog_backend.services.provider_service import ProviderService
+
+
+router = APIRouter()
+
+
+@router.post("/provider",
+             summary="Add a provider to the database",
+             tags=["admin"],
+             response_model=ProviderResponse)
+async def post_provider(provider_req: ProviderRequest,
```

Comment:
I prefer the methods that adds data to be called `add_something` instead of `post_something`.
The `post` refers to the http-method and not the effect of adding something new to a collection.